File: rust/muse/src/algorithms/number_theory/fibonacci_number.rs
pub fn iterative_procedure(mut n: i64) -> i64 {
    let mut sign = 1;

    if n < 0 {
        if (n & 1) == 0 {
            sign = -1;
        }
        n = -n;
    }

    if n < 2 {
        return n;
    }

    let mut prev = 0;
    let mut curr = 1;

    for _ in 2..=n {
        (prev, curr) = (curr, prev + curr);
    }

    sign * curr
}

pub fn recursive_procedure(n: i64) -> i64 {
    if n < 0 {
        if (n & 1) == 0 {
            return -recursive_procedure(-n);
        }
        return recursive_procedure(-n);
    }
    if n < 2 {
        return n;
    }
    recursive_procedure(n - 2) + recursive_procedure(n - 1)
}

#[cfg(test)]
pub(crate) mod tests;

File: python/muse/algorithms/number_theory/fibonacci_number.py
def iterative_procedure(n: int) -> int:
    sign = 1

    if n < 0:
        if (n & 1) == 0:
            sign = -1
        n = -n

    if n < 2:
        return n

    prev = 0
    curr = 1

    for _ in range(n - 1):
        prev, curr = curr, prev + curr

    return sign * curr


def recursive_procedure(n: int) -> int:
    if n < 0:
        val = recursive_procedure(-n)
        if (n & 1) == 0:
            return -val
        return val
    if n < 2:
        return n
    return recursive_procedure(n - 2) + recursive_procedure(n - 1)
